src/DataGrid.tsx

```diff
@@ -1236,10 +1220,36 @@ function DataGrid<R, SR, K extends Key>(
             </>
           )}
         </RowSelectionChangeProvider>
-
-        {/* render empty cells that span only 1 column so we can safely measure column widths, regardless of colSpan */}
-        {renderMeasuringCells(viewportColumns)}
       </DataGridDefaultRenderersProvider>
+
+      {/* render empty cells that span only 1 column so we can safely measure column widths, regardless of colSpan */}
+      {renderMeasuringCells(viewportColumns)}
+
+      {/* extra div is needed for row navigation in a treegrid */}
+      {hasGroups && (
```

Comment:
Move this div to the end. This way we can use a lower value of `zIndex` and it already uses `pointer-events: none`. Can you think of any issues with this change?